Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -449,6 +451,9 @@ final class PurchasesOrchestrator {
                     return try await sk2Product.purchase(options: options)
                 }
         } catch StoreKitError.userCancelled {
+            if self.systemInfo.dangerousSettings.customEntitlementComputation {
+                throw ErrorUtils.purchaseCancelledError()
+            }
```

Comment:
I'm not sure if this error would be catched by the following catches... Still had to test each of these cases to make sure it behaved as I expected